# Healthcare Insurance: Data Collection and Cleaning

## Objectives

- Document the dataset source and licence.
- Load and inspect the original data.
- Check missing values, duplicates, categories and numerical ranges.
- Investigate possible outliers.
- Apply justified cleaning transformations.
- Create age and BMI groups.
- Save a processed dataset for analysis.

## Inputs

- `data/raw/insurance.csv`

## Outputs

- `data/processed/insurance_clean.csv`

## Data source

The data was downloaded from the Healthcare Insurance dataset on Kaggle.
The dataset is published under the CC0 public-domain licence.

The original file is stored unchanged in `data/raw/`. Cleaning operations
will be applied to a copy, and the result will be stored in
`data/processed/`.

In [1]:
from pathlib import Path
import platform

import pandas as pd

project_root = Path.cwd()

if project_root.name == "jupyter_notebooks":
    project_root = project_root.parent

raw_data_path = (
    project_root
    / "data"
    / "raw"
    / "insurance.csv"
)

processed_data_path = (
    project_root
    / "data"
    / "processed"
    / "insurance_clean.csv"
)

print(f"Python version: {platform.python_version()}")
print(f"pandas version: {pd.__version__}")
print(f"Project root: {project_root}")

Python version: 3.12.8
pandas version: 2.1.1
Project root: /Users/ewa/Documents/vscode-projects/code-institute-2026/Healthcare-Insurance-Cost-Analysis


## Load the source data

Pandas is used to load the CSV file into a DataFrame. A DataFrame represents
the data as rows and columns.

In [2]:
insurance_raw = pd.read_csv(raw_data_path)

print(
    f"The dataset contains {insurance_raw.shape[0]:,} rows "
    f"and {insurance_raw.shape[1]} columns."
)

insurance_raw.head()

The dataset contains 1,338 rows and 7 columns.


,age,sex,bmi,children,smoker,region,charges
0,19,female,27.900,0,yes,southwest,16884.92400
1,18,male,33.770,1,no,southeast,1725.55230
2,28,male,33.000,3,no,southeast,4449.46200
3,33,male,22.705,0,no,northwest,21984.47061
4,32,male,28.880,0,no,northwest,3866.85520


In [3]:
insurance_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


In [4]:
insurance_raw.describe(include="all").T

,count,unique,top,freq,mean,std,min,25%,50%,75%,max
age,1338.0,NaN,NaN,NaN,39.207025,14.04996,18.0,27.0,39.0,51.0,64.0
sex,1338,2,male,676,NaN,NaN,NaN,NaN,NaN,NaN,NaN
bmi,1338.0,NaN,NaN,NaN,30.663397,6.098187,15.96,26.29625,30.4,34.69375,53.13
children,1338.0,NaN,NaN,NaN,1.094918,1.205493,0.0,0.0,1.0,2.0,5.0
smoker,1338,2,no,1064,NaN,NaN,NaN,NaN,NaN,NaN,NaN
region,1338,4,southeast,364,NaN,NaN,NaN,NaN,NaN,NaN,NaN
charges,1338.0,NaN,NaN,NaN,13270.422265,12110.011237,1121.8739,4740.28715,9382.033,16639.912515,63770.42801


In [5]:
insurance_raw.isna().sum()

age         0
sex         0
bmi         0
children    0
smoker      0
region      0
charges     0
dtype: int64

In [6]:
duplicate_count = insurance_raw.duplicated().sum()

print(f"Exact duplicate rows: {duplicate_count}")

Exact duplicate rows: 1


In [7]:
for column in ["sex", "smoker", "region"]:
    values = sorted(
        insurance_raw[column]
        .dropna()
        .unique()
    )

    print(f"{column}: {values}")

sex: ['female', 'male']
smoker: ['no', 'yes']
region: ['northeast', 'northwest', 'southeast', 'southwest']


## Initial quality findings

- The dataset contains **1338 rows and 7 columns**
- The dataset includes **3 categorical variables** ('sex', 'smoker', 'region') and **4 numerical variables** ('age', 'bmi', 'children', 'charges')
- All columns have the correct data types ('int64', 'float64', and 'object')
- The dataset contains no missing values
- The dataset contains 1 duplicate row
- The categorical values are:
    - **sex:** female, male
    - **smoker:** no, yes
    - **region:** northeast, northwest, southeast, southwest
- The numerical variables appear to have reasonable ranges:
    - **Age:** 18-64 years
    - **BMI:** approximately 15.96-53.13
    - **Children:** 0-5
    - **Charges:** approximately $1,121.87-$63,770.43
- No obvious data quality issues were indentified apart from the single duplicate record